test/jdk/java/net/httpclient/ContentLengthHeaderTest.java

```diff
@@ -97,26 +95,29 @@ public void getWithNoBody() throws IOException, InterruptedException {
                 .uri(URI.create(testContentLenURI + GET_NO_BODY_PATH))
                 .build();
         HttpClient hc = HttpClient.newHttpClient();
-        hc.send(req, HttpResponse.BodyHandlers.ofString(UTF_8));
+        HttpResponse<String> resp = hc.send(req, HttpResponse.BodyHandlers.ofString(UTF_8));
+        assertEquals(resp.statusCode(), 200,
+                "Content-length header was set in request but should not be present.");
     }
 
     @Test
     // A GET request with a request body should have a Content-length header
     public void getWithBody() throws IOException, InterruptedException {
         HttpRequest req = HttpRequest.newBuilder()
                 .version(HTTP_1_1)
-                .method("GET", HttpRequest.BodyPublishers.ofString(GET_BODY_STRING))
+                .method("GET", HttpRequest.BodyPublishers.ofString(GET_BODY))
                 .uri(URI.create(testContentLenURI + GET_BODY_PATH))
                 .build();
         HttpClient hc = HttpClient.newHttpClient();
-        hc.send(req, HttpResponse.BodyHandlers.ofString(UTF_8));
+        HttpResponse<String> resp = hc.send(req, HttpResponse.BodyHandlers.ofString(UTF_8));
+        assertEquals(resp.statusCode(), 200,
+                "Content-length header was not set in request but should be present.");
     }
 
-    public static void handleResponse(HttpExchange ex) throws IOException {
-        // Meaningless response to finish off the exchange
+    public static void handleResponse(HttpExchange ex, int rCode) throws IOException {
         try (OutputStream os = ex.getResponseBody()) {
-            byte[] bytes = responseString.getBytes(UTF_8);
-            ex.sendResponseHeaders(200, bytes.length);
+            byte[] bytes = RESPONSE_BODY.getBytes(UTF_8);
```

Comment:
IRL you would have a different response body for success and error but since we don't care about the response body in this test it's ok to send the same.